test/new_relic/gem_notifications_tests.rb

```diff
@@ -0,0 +1,73 @@
+# encoding: utf-8
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+
+require_relative '../test_helper'
+require_relative '../../.github/workflows/scripts/slack_gem_notifications/notifications_methods'
+
+class GemNotifications < Minitest::Test
+  def invalid_gem_response
+    response = MiniTest::Mock.new
+    response.expect :success?, false
+  end
+
+  def valid_gem_response
+    response = MiniTest::Mock.new
+    response.expect :success?, true
+  end
+
+  def http_get_response
+    [{"created_at": "2022-03-30T16:15:29.083Z", "platform": "ruby"},
+      {"created_at": "2021-03-30T16:15:29.083Z", "platform": "java"},
+      {"created_at": "2020-03-30T16:15:29.083Z", "platform": "ruby"}]
+  end
+
+  def test_check_for_updates_if_empty
+    response = MiniTest::Mock.new
+    response.expect :empty?, true
+    [].stub :empty?, response do
+      error = assert_raises(Exception) { check_for_updates([]) }
+      assert_equal("Nothing to see here! watched_gems cannot be empty", error.message)
+    end
+  end
+
+  def valid_gem_name
+    response = valid_gem_response()
+    HTTParty.stub :get, response do
+      assert_nil is_valid_gem?("puma!")
```

Comment:
I'm confused by this one. If a valid gem response is being returned, then shouldn't the `is_valid_gem?` call return the response object instead of `nil`?

Reply:
I was stumped on how this was passing. Finally figured out the method didn't start with `test` 🙃